File: run_kde.py
cell_lines_file=open("cellList.txt")
cell_lines=cell_lines_file.readlines()
cell_lines.append("ESC") 

import sys
import os
path="/srv/gs1/projects/snyder/imk1/P01Project/ChIPseqData/"
outputpath=path+"quest/"
i=int(sys.argv[1])-1

marks=["H3K27ac", "H3K27me3", "H3K4me1", "H3K4me3", "p300"] #, "Pol2"]
#marks=["background"]

cell_line_index=i/len(marks)
mark_index=i%len(marks)
mark=marks[mark_index]

cell_line=cell_lines[cell_line_index].strip()

os.system("/srv/gs1/software/R/R-3.0.1/bin/Rscript run_kde.R %s %s" % (mark,cell_line) )

File: run_kde.R
args=commandArgs(trailingOnly=T)
mark=args[1]
cell.line=args[2]
load("hg18.RData") 
load("kernels.RData") 
source("ChIP.R")
in.path="/srv/gs1/projects/snyder/imk1/P01Project/ChIPseqData/quest/"
out.path="/srv/gs1/projects/snyder/imk1/P01Project/kdes/"
Step=100
width=10000
Genome<-generate.genome.table(A,Step=Step)
a<-load.reads(paste(in.path,cell.line,"_","background",".quest",sep=""))
SHIFT<-calculate.shift(a,Genome)
print(SHIFT)
if (SHIFT>0)
   a<-shift(a,SHIFT)
KRN<-generate.kernel(name=mark,step=Step,width=width,source=kernels)
data.out<-ChIP(a,KRN,Step=Step,Genome=Genome)
save(data.out, file=paste(out.path,"/",cell.line,"_",mark,"_background.RData",sep=""))


